[v0.1.0 regression] compilation fails with GOOS=freebsd

<!--
Hello! To ensure this issue is correctly addressed as soon as possible by the IPFS team, please try to make sure:

- This issue is relevant to this repository's topic or codebase.

- A clear description is provided. It should includes as much relevant information as possible and clear scope for the issue to be actionable.

FOR GENERAL DISCUSSION, HELP OR QUESTIONS, please see the options at https://ipfs.io/help or head directly to https://discuss.ipfs.io.

(you can delete this section after reading)
-->

go-ipfs-files@v0.1.0 can't be compiled for `GOOS=freebsd`:
```
$ GOOS=freebsd go build
# github.com/ipfs/go-ipfs-files
./filewriter.go:25:13: undefined: createNewFile
./filewriter.go:30:10: cannot assign error to err in multiple assignment
./filewriter.go:47:6: undefined: isValidFilename
```


v0.0.9 was compilable.

Go version: 1.17.6

--- a/filewriter_unix.go
+++ b/filewriter_unix.go
@@ -1,5 +1,5 @@
-//go:build darwin || linux || netbsd || openbsd
-// +build darwin linux netbsd openbsd
+//go:build darwin || linux || netbsd || openbsd || freebsd
+// +build darwin linux netbsd openbsd freebsd
 
 package files
 

